docs/decisions/00XX-realtime-api-clients.md

```diff
@@ -0,0 +1,206 @@
+- Con:
+  - overkill for simple control events
+
+### 3. Treat everything as events
+This would mean that all events are retained and returned to the developer as is, without any transformation.
+
+- Pro:
+  - no transformation needed
+  - easy to maintain
+- Con:
+  - nothing easing the burden on the developer, they need to deal with the raw events
+  - no way to easily switch between one provider and another
+
+## Decision Outcome - Content and Events
+
+Chosen option: ...
+
+# Programming model
+
+## Considered Options - Programming model
+The programming model for the clients needs to be simple and easy to use, while also being able to handle the complexity of the realtime api's.
+
+_In this section we will refer to events for both content and events, regardless of the decision made in the previous section._
+
+1. Async generator for receiving events, that yields contents, combined with a event handler/callback mechanism for receiving events and a function for sending events
+   - 1a: Single event handlers, where each event is passed to the handler
+   - 1b: Multiple event handlers, where each event type has its own handler
+2. Event buffers/queues that are exposed to the developer, start sending and start receiving methods, that just initiate the sending and receiving of events and thereby the filling of the buffers
+
+### 1. Async generator for receiving events, that yields contents, combined with a event handler/callback mechanism for receiving events and a function for sending events
+This would mean that the client would have a mechanism to register event handlers, and the integration would call these handlers when an event is received. For sending events, a function would be created that sends the event to the service.
+
+- Pro:
+  - without any additional setup you get content back, just as with "regular" chat models
+  - event handlers are mostly for more complex interactions, so ok to be slightly more complex
+- Con:
+  - developer judgement needs to be made (or exposed with parameters) on what is returned through the async generator and what is passed to the event handlers
+
+### 2. Event buffers/queues that are exposed to the developer, start sending and start receiving methods, that just initiate the sending and receiving of events and thereby the filling of the buffers
+This would mean that the there are two queues, one for sending and one for receiving, and the developer can listen to the receiving queue and send to the sending queue. Internal things like parsing events to content types and auto-function calling are processed first, and the result is put in the queue, the content type should use inner_content to capture the full event and these might add a message to the send queue as well.
+
+- Pro:
+  - simple to use, just start sending and start receiving
+  - easy to understand, as queues are a well known concept
+  - developers can just skip events they are not interested in
+- Con:
```

Comment:
If the audio is large or high volume, we would have noticeable lag with a queue approach, right?